### PR TITLE
New endpoint to verify an account's billing information

### PR DESCRIPTION
**Pull request by joannasese**
If the account has billing information for anything other than a credit card, this will return 422, with an appropriate error message. If successful, this will return 200, with the transaction's data.
`gateway_code` (optional) is the code for the gateway to use for the verification. If unspecified, a gateway will be selected using the normal rules.

Example
```c#
// verify without specifying gateway code to use 
billingInfo.Verify()
// verify with a specific gateway code
billingInfo.Verify("gateway-code")
```

### CODE REVIEW

File: Library/VerifyGateway.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Xml;
+
+namespace Recurly
+{
+    class VerifyGateway : RecurlyEntity
```
Comment (douglasmiller) [inline Library/VerifyGateway.cs:6]: Would `VerifyBillingInfo` be a more accurate name?

File: Library/VerifyGateway.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+              if (reader.Name == "gateway_code")
+                    GatewayCode = reader.ReadElementContentAsString();
+            }
+        }
+
+        internal override void WriteXml(XmlTextWriter writer)
+        {
+            writer.WriteStartElement("verify");
+
+            writer.WriteElementString("gateway_code", GatewayCode);
```
Comment (douglasmiller) [inline Library/VerifyGateway.cs:33]: I think that if you only include this if `GatewayCode` is not null/empty, then the conditional checks in the `Verify` method can be removed entirely.

File: Library/VerifyGateway.cs (hunk 1 above)
Comment (joannasese) [reply Library/VerifyGateway.cs:6]: probably.

File: Library/VerifyGateway.cs (hunk 2 above)
Comment (joannasese) [reply Library/VerifyGateway.cs]: Following up on discussion, you'll see some adjustments. Looks much better.

**Review by douglasmiller — APPROVED** [review]
👍

**Pull request merged**